src/foam/u2/view/EditColumnsView.js

```diff
@@ -43,74 +35,58 @@ foam.CLASS({
       margin-top: 5px;
     }
   `,
-  
+
   methods: [
     function initE() {
-      // if ( this.displaySorted ) {
-      //   // TODO: How should this block be tested?
-      //   var props = this.properties;
-      //   props = this.properties.slice();
-      //   props.sort(function(a, b) {
-      //     return a.label.toLowerCase().compareTo(b.label.toLowerCase());
-      //   });
-      // } else { ...
-
-      this.addClass(this.myClass());
-
-      this.selected = []
-
-      for ( var i = 0 ; i < this.columns_.length ; i++ ) {
-        var cb = this.CheckBox.create({
-          label: this.columns_[i].label,
-          data: true
+      var index = 0;
+      this
+        .addClass(this.myClass())
+        .forEach(this.columns, function(columnName) {
+          var axiom = this.table.getAxiomByName(columnName);
+          var localStorageKey = this.table.id + '.' + columnName;
+          var valueFromLocalStorage = localStorage.getItem(localStorageKey);
+
+          // This will be true only when seeing a new model for the first time.
+          if ( valueFromLocalStorage === null ) {
```

Comment:
Currently I have it set up so that there's always a value in localStorage. Another way we could do it is to only set a value in localStorage if the checkbox has been unchecked, and remove the entry in localStorage when it's checked again. So we'd only be storing the properties that we don't want to show.

I wasn't sure which way was better, but I'm happy to switch it if anyone thinks the other way is better.

Reply:
So when a new property is added to the model tableColumns[], they will not see it.  And this is regardless if they have explicitly made a selection or not.  
Given this, I would prefer the negative case, where we record the 'not-selected'. Then they are aware of new properties we add and can consciously disable them. 
On the flip side, not-showing what's new might help testing - if they were validating what is shown,  but this would be a quick update - to deselect and run the test again.

Reply:
> So when a new property is added to the model tableColumns[], they will not see it.

That wasn't the intention, and I just tested and that's not happening for me.

Reply:
Ok, if that's not the case, then I'm fine with the currently coded approach.

Reply:
Updated to the other approach anyway since it requires less code and less space in localStorage.